packages/http/src/router/matchBaseUrl.ts

```diff
@@ -1,23 +1,55 @@
+import * as E from 'fp-ts/lib/Either';
+import * as O from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+
 import { INodeVariable, IServer } from '@stoplight/types';
 import { MatchType } from './types';
 
 const variableRegexp = /{(.*?)}/g;
 
-export function matchBaseUrl(server: IServer, baseUrl: string): MatchType {
-  const templateMatchResult = matchRequestUrlToTemplateUrl(baseUrl, server.url, server.variables);
+export function matchBaseUrl(server: IServer, baseUrl: string) {
```

Comment:
Let's always try to be explicit with the return types, especially when they're monads

```suggestion
export function matchBaseUrl(server: IServer, baseUrl: string): E.Either<Error, MatchType> {
```

Reply:
This - and the second one - was actually intentional. I was looking to commit this in 2 phases, first just changing the internals, but keeping the behavior unchanged, then - if you are OK with the implementation - go ahead and bubble the signature change up. So this change is comming.:) (See **Status**.)